include/net/dns_resolve.h

```diff
@@ -62,6 +62,19 @@ enum dns_query_type {
 #define MDNS_SERVER_COUNT 0
 #endif /* CONFIG_MDNS_RESOLVER */
 
+/* If LLMNR is enabled, then add some extra well known multicast servers to the
+ * server list.
+ */
+#if defined(CONFIG_LLMNR_RESOLVER)
+#if defined(CONFIG_NET_IPV6) && defined(CONFIG_NET_IPV4)
+#define LLMNR_SERVER_COUNT 2
+#else
+#define LLMNR_SERVER_COUNT 1
+#endif /* CONFIG_NET_IPV6 && CONFIG_NET_IPV4 */
+#else
+#define LLMNR_SERVER_COUNT 0
+#endif /* CONFIG_MDNS_RESOLVER */
+
```

Comment:
Add a #define DNS_MAX_SERVERS (or something semantically better if you have any idea) which will add the 3 of the server count. Will be easier in relevant code then, no need to make the addition manytimes.

Reply:
@tbursztyka , do you mean instead of:

~~~
#define DNS_MAX_MCAST_SERVERS (MDNS_SERVER_COUNT + LLMNR_SERVER_COUNT)
} servers[CONFIG_DNS_RESOLVER_MAX_SERVERS + DNS_MAX_MCAST_SERVERS];
~~~

to have:

~~~
#define DNS_MAX_SERVERS (CONFIG_DNS_RESOLVER_MAX_SERVERS + MDNS_SERVER_COUNT + LLMNR_SERVER_COUNT)
} servers[DNS_MAX_SERVERS];
~~~

? (The comment looks misplaced a bit.)

I'm actually not sure which way is better. I'd say, I'd prefer that whenever possible, CONFIG_* options were used directly in various expression, instead of hidden by extra layer of indirection (easier to grep code this way).

Reply:
The idea, as you saw in the code, is to avoid repeating the lengthy addition of all these CONFIG_ yes.